src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -150,6 +149,70 @@ const BridgeConfirmation: FC = () => {
     }
   }, [env, formData, getTokenPrice, callIfMounted, tokens]);
 
+  const onApprove = () => {
+    if (connectedProvider && account.status === "successful") {
+      setApprovalTask({ status: "loading" });
+      void approve({
+        token,
+        owner: account.data,
+        spender: from.contractAddress,
+        provider: connectedProvider.provider,
```

Comment:
Here you're assuming that the network selected in `from` matches the network of the connectedProvider and that might not be the case. If Metamask is connected to Polygon zkEVM and the user has selected ethereum in the `from` selector this should fail since we would be sending the wrong `contractAddress`.

At some point (maybe in the adapter?) we should make sure that `from.chainId === connectedProvider.chainId` [as the bridge function does](https://github.com/0xPolygonHermez/zkevm-bridge-ui/blob/develop/src/contexts/bridge.context.tsx#L836-L844), and ask for a network change when required.

Reply:
I forgot about this check, fixed!